packages/zowe-explorer/src/dataset/actions.ts

```diff
@@ -931,20 +931,28 @@ export async function submitJcl(datasetProvider: api.IZoweTree<api.IZoweDatasetT
     const profiles = Profiles.getInstance();
     let sessProfileName;
     if (regExp === null) {
-        const profileNamesList = ProfileManagement.getRegisteredProfileNameList(globals.Trees.JES);
-        if (profileNamesList.length) {
-            const quickPickOptions: vscode.QuickPickOptions = {
-                placeHolder: localize("submitJcl.qp.placeholder", "Select the Profile to use to submit the job"),
-                ignoreFocusOut: true,
-                canPickMany: false,
-            };
-            sessProfileName = await api.Gui.showQuickPick(profileNamesList, quickPickOptions);
-            if (!sessProfileName) {
-                api.Gui.infoMessage(localizedStrings.opCancelled);
-                return;
+        if (!doc.uri.fsPath.includes(globals.ZOWETEMPFOLDER)) {
+            const profileNamesList = ProfileManagement.getRegisteredProfileNameList(globals.Trees.JES);
+            if (profileNamesList.length > 1) {
+                const quickPickOptions: vscode.QuickPickOptions = {
+                    placeHolder: localize("submitJcl.qp.placeholder", "Select the Profile to use to submit the job"),
+                    ignoreFocusOut: true,
+                    canPickMany: false,
+                };
+                sessProfileName = await api.Gui.showQuickPick(profileNamesList, quickPickOptions);
+                if (!sessProfileName) {
+                    api.Gui.infoMessage(localizedStrings.opCancelled);
+                    return;
+                }
+            } else if (profileNamesList.length > 0) {
+                sessProfileName = profileNamesList[0];
+            } else {
+                api.Gui.showMessage(localize("submitJcl.noProfile", "No profiles available"));
             }
         } else {
-            api.Gui.showMessage(localize("submitJcl.noProfile", "No profiles available"));
+            // eslint-disable-next-line no-useless-escape
```

Comment:
This linter exception can be removed since the RegEx expression was replaced 👍

Reply:
Hey thanks Trae for noticing that